src/SmartTransactionsController.ts

```diff
@@ -889,4 +889,56 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
       );
     });
   }
+
+  #wipeSmartTransactionsPerChainId(
+    chainId: Hex,
+    selectedAddressLowerCase: string,
+  ): void {
+    const { smartTransactions } = this.state.smartTransactionsState;
+    const smartTransactionsForSelectedChain: SmartTransaction[] =
+      smartTransactions?.[chainId];
+    if (
+      !smartTransactionsForSelectedChain ||
+      smartTransactionsForSelectedChain.length === 0
+    ) {
+      return;
+    }
+    const newSmartTransactionsForSelectedChain =
+      smartTransactionsForSelectedChain.filter(
+        (smartTransaction: SmartTransaction) =>
+          smartTransaction.txParams?.from !== selectedAddressLowerCase,
+      );
+    this.update({
+      smartTransactionsState: {
+        ...this.state.smartTransactionsState,
+        smartTransactions: {
+          ...smartTransactions,
+          [chainId]: newSmartTransactionsForSelectedChain,
+        },
+      },
+    });
+  }
+
+  wipeSmartTransactions(
+    selectedAddress?: string,
```

Comment:
Do we need to call this `selectedAddress` or would `address` be simpler since whether it is selected could be domain knowledge this method doesn't need to know?